mlflow/recipes/steps/train.py

```diff
@@ -1121,8 +1138,64 @@ def _safe_dump_with_numeric_values(self, data, file, **kwargs):
                 processed_data[key] = float(value)
             elif isinstance(value, np.integer):
                 processed_data[key] = int(value)
+            elif isinstance(value, dict):
+                processed_data[key] = str(value)
             else:
                 processed_data[key] = value
 
         if len(processed_data) > 0:
             yaml.safe_dump(processed_data, file, **kwargs)
+
+    def _rebalance_classes(self, train_df):
+        import pandas as pd
+        import numpy as np
+        from sklearn.utils.class_weight import compute_class_weight
+
+        resampling_minority_percentage = self.step_config.get(
+            "resampling_minority_percentage", _REBALANCING_DEFAULT_RATIO
+        )
+
+        df_positive_class = train_df[train_df[self.target_col] == self.positive_class]
+        df_negative_class = train_df[train_df[self.target_col] != self.positive_class]
+
+        classes = np.unique(train_df[self.target_col])
+        class_weights = compute_class_weight(
+            class_weight="balanced",
+            classes=classes,
+            y=train_df[self.target_col],
+        )
+
+        self.original_class_weights = dict(zip(classes, class_weights))
```

Comment:
Can we factor this section out, compute class weights irrespective whether rebalance is enabled? It's a useful metric to show in step card regardless.